lte/gateway/python/integ_tests/s1aptests/test_attach_detach_secondary_pdn_multi_ue.py

```diff
@@ -0,0 +1,112 @@
+"""
+Copyright (c) 2016-present, Facebook, Inc.
+All rights reserved.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree. An additional grant
+of patent rights can be found in the PATENTS file in the same directory.
+"""
+
+import unittest
+import time
+
+import s1ap_types
+import s1ap_wrapper
+
+
+class TestSecondaryPdnConnReqMultiUe(unittest.TestCase):
+    def setUp(self):
+        self._s1ap_wrapper = s1ap_wrapper.TestWrapper()
+
+    def tearDown(self):
+        self._s1ap_wrapper.cleanup()
+
+    def test_seconday_pdn_conn_req_multi_ue(self):
+        """ attach/detach + PDN Connectivity Requests with 4 UEs """
+        num_ues = 4
+        ue_ids = []
+        bearer_ids = []
+        self._s1ap_wrapper.configUEDevice(num_ues)
+        for _ in range(num_ues):
+            req = self._s1ap_wrapper.ue_req
+            ue_id = req.ue_id
+            print(
+                "******************* Running End to End attach for UE id ",
+                ue_id,
+            )
+            # Attach
+            self._s1ap_wrapper.s1_util.attach(
+                ue_id,
+                s1ap_types.tfwCmd.UE_END_TO_END_ATTACH_REQUEST,
+                s1ap_types.tfwCmd.UE_ATTACH_ACCEPT_IND,
+                s1ap_types.ueAttachAccept_t,
+            )
+
+            # Wait on EMM Information from MME
+            self._s1ap_wrapper._s1_util.receive_emm_info()
+            ue_ids.append(ue_id)
+
+        self._s1ap_wrapper._ue_idx = 0
+        for i in range(num_ues):
```

Comment:
Code checker generated the following error:

" Loop control variable 'i' not used within the loop body. If this is intended, start the name with an underscore."